Warning during zeek builds

This is coming up during a Zeek build. I haven't looked at the offending code at all. I'm seeing it on macOS but I think it happens on ubuntu as well.

```
/Users/tim/Desktop/projects/zeek-master/auxil/spicy/3rdparty/justrx/src/tests/retest.c:212:48: warning: result of comparison of constant -1 with expression of type 'char' is always true [-Wtautological-constant-out-of-range-compare]
  212 |     while ( (c = getopt(argc, argv, "adlit:")) != -1 ) {
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~
1 warning generated.
```

--- a/src/tests/retest.c
+++ b/src/tests/retest.c
@@ -208,7 +208,7 @@ int main(int argc, char** argv) {
     const char* d;
 
     // Parse command line options with getopt().
-    char c;
+    int c;
     while ( (c = getopt(argc, argv, "adlit:")) != -1 ) {
         switch ( c ) {
             case 'a': cflags |= REG_ANCHOR; break;

